spec/em/em_spec.rb

```diff
@@ -65,6 +65,21 @@
       end.to raise_error('some error')
     end
 
+    it "should timeout if we wait longer than :read_timeout" do
+      expect do
+        EM.run do
+          client = Mysql2::EM::Client.new DatabaseCredentials['root'].merge(read_timeout: 1)
+          defer = client.query "SELECT sleep(2)"
+          defer.callback do
+            # This _shouldn't_ be run, but it needed to prevent the specs from
+            # freezing if this test fails.
+            client.close
+            EM.stop_event_loop
+          end
+        end
+      end.to raise_error(Mysql2::EM::ReadTimeout)
```

Comment:
Can you switch to the `defer.errback` pattern see in the tests below? It is problematic to throw exceptions outside the scope of `EM.run` (throwing exceptions like this in fact would crash EM 1.2.x until I finally finally figured out the problem in EM 1.2.6)

Reply:
Sure, I'll switch to `defer.errback` :)